### DISCUSSION TITLE
Failing builds (including CMake) on (standard) Manjaro Linux

### DISCUSSION THREAD

**Discussion by ppenguin** [apohl79/audiogridder#306] [Help]
Started [here](https://github.com/apohl79/audiogridder/issues/298).
Using @apohl79 's [comment here](https://github.com/apohl79/audiogridder/issues/298#issuecomment-754788671) with some necessary changes, I arrived at the following: [my fork](https://github.com/apohl79/audiogridder/compare/master...ppenguin:fix-manjaro-cmake-build).

Note that the original `FindFFmpeg.cmake` didn't find my `libav*` (`ffmpeg`), though it is complete and installed in the standard location. My modification using `pkgconfig` found the libs, but I still get linker errors at least partly related to this.

Also, the standard method of finding `JUCE` (`find_package(JUCE CONFIG REQUIRED)`) didn't work for me, because my `JUCE` comes without `juce-config.cmake`, though appears complete otherwise. But with my mods all `JUCE`-related issues appear to be solved.

I'm a noob regarding `cmake`, so maybe someone has ideas?

**Comment by apohl79** [comment]
The orig FindFFmpeg.cmake looks for static libs (*.a files) only. Your installation might just ship shared objects. If the static ones are just not found, you could add the location to the cmake file.

You could just checkout JUCE (https://github.com/juce-framework/JUCE.git) into the AG folder and use it (means you need no find_package but add_subdirectory). As AG links all juce code into the binary, your system wide installation would not cause issues.

**Reply by ppenguin** [reply]
Thanks for the reply.

> But with my mods all JUCE-related issues appear to be solved.

(That was fixed by changing `add_subdirectory(JUCE)` to `add_subdirectory(${JUCE_DIR} JUCE)` and setting `${JUCE_DIR}` to `"/usr/share/juce"`)

The remaining issues appear not to be `JUCE` related, they are linking errors seemingly from 3 causes:

```
/usr/bin/ld: /usr/bin/ld: DWARF error: could not find variable specification at offset 58b07
/usr/bin/ld: DWARF error: could not find variable specification at offset 58b14
...
AudioGridderFx_artefacts/RelWithDebInfo/libAudioGridder_SharedCode.a(PluginProcessor.cpp.o):(.bss+0x0): first defined here
/usr/bin/ld: AudioGridderFx_artefacts/RelWithDebInfo/libAudioGridder_SharedCode.a(CoreDump.cpp.o):(.bss+0x2): multiple definition of `e4
7::Defaults::getLogTagSource'; AudioGridderFx_artefacts/RelWithDebInfo/libAudioGridder_SharedCode.a(PluginProcessor.cpp.o):(.bss+0x0): f
irst defined here
...
/usr/bin/ld: AudioGridderFx_artefacts/RelWithDebInfo/libAudioGridder_SharedCode.a(ImageReader.cpp.o): in function `e47::ImageReader::ini
tCodec()':
/home/jeroen/devel/github.com/ppenguin/audiogridder/Plugin/Source/ImageReader.cpp:97: undefined reference to `avcodec_find_decoder_by_na
me'
/usr/bin/ld: /home/jeroen/devel/github.com/ppenguin/audiogridder/Plugin/Source/ImageReader.cpp:103: undefined reference to `av_malloc'
/usr/bin/ld: /home/jeroen/devel/github.com/ppenguin/audiogridder/Plugin/Source/ImageReader.cpp:108: undefined reference to `av_init_pack
et'
```

There's a lot of them, of which the first type is new to me, the second type is somehow related to AG/make, and the last type appears to be related to `ffmpeg` (so an issue with my alternative to `FindFFmpeg`, even though it says it can find it during configure).

Does the issue with static libs mean that AG *can* only build against static ffmpeg libs, or could it be made to work "normally" by fixing the CMake further down the line?
